test/jobs/job-sharing.ts

```diff
@@ -0,0 +1,225 @@
+import { expect } from 'chai';
+import { describe, it, before } from 'mocha';
+import { JobStatus } from 'models/job';
+import hookServersStartStop from '../helpers/servers';
+import { hookStacCatalog, hookStacItem } from '../helpers/stac';
+import { hookTransaction } from '../helpers/db';
+import { hookJobStatus, buildJob } from '../helpers/jobs';
+
+const collectionWithEULAFalseAndGuestReadTrue = 'C1233800302-EEDTEST';
+const collectionWithEULATrueAndGuestsReadTrue = 'C1233860183-EEDTEST';
+const collectionWithEULAFalseAndGuestReadFalse = 'C1233147317-EEDTEST';
+const collectionWithEULANonexistent = 'C1234088182-EEDTEST';
+
+const baseJobProperties = {
+  numInputGranules: 5,
+  links: [{
+    href: 's3://example-bucket/public/example/path1.tif',
+    type: 'image/tiff',
+    rel: 'data',
+    bbox: [-10, -10, 10, 10],
+    temporal: {
+      start: new Date('2020-01-01T00:00:00.000Z'),
+      end: new Date('2020-01-01T01:00:00.000Z'),
+    },
+  },
+  {
+    href: 's3://example-bucket/public/example/path2.tif',
+    type: 'image/tiff',
+    rel: 'data',
+    bbox: [-10, -10, 10, 10],
+    temporal: {
+      start: new Date('2021-01-01T00:00:00.000Z'),
+      end: new Date('2021-01-01T01:00:00.000Z'),
+    },
+  }],
+  request: 'http://example.com/harmony?job=completedJob',
+  status: JobStatus.SUCCESSFUL,
+};
+
+const jobWithNoCollections = buildJob({
+  username: 'joe',
```

Comment:
Maybe define:
```
const jobOwner = 'joe';
const notJobOwner = 'jill';
```
and then use those constants in all the tests so you can tell from just looking at a single line the tests are trying to access a job that isn't theirs?

Only commenting because when I saw 'jill' in the tests I wasn't clear whether or not they owned the job.

By the way these tests were really easy to follow otherwise - nice job with creating all the jobs upfront with easy to understand names.

Reply:
Thanks! Yea that makes sense. I updated it here: https://github.com/nasa/harmony/commit/7f57a795a039e8d9e837911aeda28ae41db4608d